Values are never used after being assigned

I'm a big fan of static analysis tools like [honnef.co/go/tools/cmd/staticcheck](https://github.com/dominikh/go-tools/tree/master/cmd/staticcheck)

Running this tool, I found 2 unchecked errors:

```
solr_sync.go:48:6: this value of shouldReconnect is never used (SA4006)
solr_sync.go:67:6: this value of shouldReconnect is never used (SA4006)
```

The issue can be seen here for [solr_sync.go:48](https://github.com/sendgrid/go-solr/blob/master/solr_sync.go#L48-L50) and also here for [solr_sync.go:67](https://github.com/sendgrid/go-solr/blob/master/solr_sync.go#L67-L70)

The `continue` statement continues the **for** loop.

Maybe these `continue` statements should be **`break`** statements to terminate execution of the `select` statement so the flow of control can actually reach the `if` statement checking the `shouldReconnect` variable on [line 85](https://github.com/sendgrid/go-solr/blob/master/solr_sync.go#L85).

--- a/solr_sync.go
+++ b/solr_sync.go
@@ -2,8 +2,9 @@ package solr
 
 import (
 	"fmt"
-	"github.com/samuel/go-zookeeper/zk"
 	"time"
+
+	"github.com/samuel/go-zookeeper/zk"
 )
 
 func (s *solrZkInstance) Listen() error {
@@ -47,7 +48,7 @@ func (s *solrZkInstance) Listen() error {
 					logErr(cEvent.Err)
 					shouldReconnect = isConnectionClosed(err)
 					sleepTime = backoff(sleepTime)
-					continue
+					break
 				}
 				// do something if its not a session or disconnect
 				if cEvent.Type == zk.EventNodeDataChanged {
@@ -67,7 +68,7 @@ func (s *solrZkInstance) Listen() error {
 					shouldReconnect = isConnectionClosed(err)
 					log.Error(fmt.Errorf("[go-solr] error on nevent %v", nEvent))
 					sleepTime = backoff(sleepTime)
-					continue
+					break
 				}
 				// do something if its not a session or disconnect
 				if nEvent.Type == zk.EventNodeDataChanged || nEvent.Type == zk.EventNodeChildrenChanged {

